src/context.rs

```diff
@@ -26,12 +26,12 @@ use ffi::enums::{Status, Antialias, LineCap, LineJoin, FillRule};
 use ::patterns::{wrap_pattern, Pattern};
 use surface::Surface;
 
-pub struct RectangleVec {
+pub struct RectangleVec<'a> {
     ptr: *mut cairo_rectangle_list_t,
-    pub rectangles: CVec<Rectangle>,
+    pub rectangles: &'a[Rectangle], // Vec<Rectangle>,
 }
```

Comment:
What do you mean? I thought that we freed RectangleVec ourselves.

Also, RectangleVec.rectangles is created at the same time as the RectangleVec instance and also freed at the same time as the RectangleVec instance. In short, RectangleVec and RectangleVec.rectangles have the same lifetime.

Am I wrong in this? Or do you mean that the RectangleVec instance can be copied? If so, that's bad...